server/controllers/foodCategory.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-console */
+const Boom = require('@hapi/boom');
+const { getFoodCategory } = require('../database/queries');
+
+const foodCategory = async (req, res, next) => {
+  const { user } = req;
+  const { categoryId } = req.params;
+
+  try {
+    if (!user) {
+      throw Boom.unauthorized('you are not logged in');
+    }
+    const { rows, rowCount } = await getFoodCategory(user.id, categoryId);
+    if (rowCount === 0) {
+      res
+        .status(200)
+        .json({ status: 200, data: null, message: 'no food to show' });
```

Comment:
I actually think it's a little better to have this controller return `data: []` instead of `data: null` if there are no rows.  The frontend then only has to deal with one type (arrays) instead of two, and it can safely do a `map` or `forEach` on the empty array when it renders food items without having to check for `null` first.

Also, I think you can remove `message` from this endpoint response, both here and on line 22 below.  The frontend can create both of those strings by looking at the length of the data array, and in general the less user-facing text you have coming from the backend the better, because it's more complicated to localize those if you were making a multi-lingual app

Reply:
this actually makes more sense to work with, thank you so much

Reply:
👍